File: core/src/main/java/com/a65apps/core/interactors/contacts/PersonDetailsRepository.java
package com.a65apps.core.interactors.contacts;

import com.a65apps.core.entities.Contact;
import com.a65apps.core.entities.Person;

import java.util.List;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Single;

public interface PersonDetailsRepository {
    Single<List<Contact>> getContactsByPerson(@NonNull final String personId);
    Single<Person> getPersonDetails(@NonNull final String personId);
}

File: core/src/main/java/com/a65apps/core/interactors/contacts/PersonDetailsRepository.kt
package com.a65apps.core.interactors.contacts

import com.a65apps.core.entities.Contact
import com.a65apps.core.entities.Person
import kotlinx.coroutines.flow.Flow

interface PersonDetailsRepository {
    fun getContactsByPerson(personId: String): Flow<List<Contact>>
    fun getPersonDetails(personId: String): Flow<Person>
}